src/com/selftest/stack/StackImpl.java

```diff
@@ -0,0 +1,59 @@
+
+package com.selftest.stack;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class StackImpl<T> implements Stack<T> {
+
+	private T[] stack;
+	private int size;
+
+	@SafeVarargs
+	public StackImpl( int size, T ... values ) {
+		stack = Arrays.copyOf( values, size );
+		size = 0;

```

Comment:
If the user were to add values to the constructor, would they expect this to discard them?